diesel/src/sqlite/connection/mod.rs

```diff
@@ -129,6 +135,59 @@ impl SqliteConnection {
             Statement::prepare(&self.raw_connection, sql)
         })
     }
+
+    /// Expose a function to SQL
+    pub fn create_scalar_function<F, T>(
+        &mut self,
+        fn_name: &str,
+        n_arg: libc::c_int,
+        deterministic: bool,
+        x_func: F
+    ) -> QueryResult<()>
+    where
+        F: FnMut(&Context) -> T,
+        T: IntoSqliteResult
+    {
+        // create_scalar_function is translated from rusqlite
+
+        use self::app_defined_fn::{call_boxed_closure, free_boxed_value};
+
+        // https://www.sqlite.org/c3ref/create_function.html
+        // >The length of the name is limited to 255 bytes in a UTF-8 representation, excluding the zero-terminator.
+        assert!(fn_name.as_bytes().len() <= 255);
```

Comment:
This is a programmer error, right, as it's only dependent on the function name size? Can you add a custom error message?